shoop/utils/i18n.py

```diff
@@ -30,14 +31,33 @@ def get_babel_locale(locale_string):
     return babel.Locale.parse(locale_string, "-")
 
 
-def get_current_babel_locale():
+def get_current_babel_locale(fallback='en_US_POSIX'):
     """
     Get a Babel locale based on the thread's locale context.
 
+    :param fallback:
+      Locale to fallback to; set to None to raise an exception instead.
     :return: Babel Locale
     :rtype: babel.Locale
     """
-    return get_babel_locale(locale_string=translation.get_language())
+    locale = get_babel_locale(locale_string=translation.get_language())
+    if not locale:
+        if fallback:
+            locale = babel.Locale(fallback)
```

Comment:
Is babel.Locale expensive? I thought it's the parsing that's expensive. Hmm.. gotta check..